spring-petclinic-api-gateway/src/test/java/org/springframework/samples/petclinic/api/application/VisitsServiceClientTest.java

```diff
@@ -0,0 +1,86 @@
+package org.springframework.samples.petclinic.api.application;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.client.circuitbreaker.EnableCircuitBreaker;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.EnableAspectJAutoProxy;
+import org.springframework.http.MediaType;
+import org.springframework.samples.petclinic.api.dto.VisitDetails;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.client.MockRestServiceServer;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.requestTo;
+import static org.springframework.test.web.client.response.MockRestResponseCreators.withServerError;
+import static org.springframework.test.web.client.response.MockRestResponseCreators.withSuccess;
+
+@ExtendWith(SpringExtension.class)
+@EnableCircuitBreaker
+@EnableAspectJAutoProxy
+@ContextConfiguration(classes = {VisitsServiceClient.class})
+class VisitsServiceClientTest {
+
+    @Autowired
+    private VisitsServiceClient visitsServiceClient;
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    private MockRestServiceServer mockServer;
+
+    @Configuration
+    static class Config {
+        @Bean
+        public RestTemplate restTemplate() {
+            return new RestTemplate();
+        }
+    }
+
+
+    @BeforeEach
+    public void setUp() {
+        mockServer = MockRestServiceServer.createServer(restTemplate);
+    }
+
+
+    @Test
+    public void getVisitsForPets_withAvailableVisitsService() {
+        mockServer.expect(requestTo("http://visits-service/pets/visits?petId=1"))
+            .andRespond(withSuccess("{\"items\":[{\"id\":5,\"date\":\"2018-11-15\",\"description\":\"test visit\",\"petId\":1}]}", MediaType.APPLICATION_JSON));
+
+        Map<Integer, List<VisitDetails>> visits = visitsServiceClient.getVisitsForPets(Collections.singletonList(1));
+
+        assertNotNull(visits);
+        assertEquals(1, visits.size());
+        assertNotNull(visits.get(1));
+        assertEquals(1, visits.get(1).size());
+        assertEquals("test visit", visits.get(1).get(0).getDescription());
```

Comment:
I would rather extract it to custom private asserion method and extract constant/variable: assertDescriptionEquals(visits.get(PET_ID) ‚’’test visit’’)

Reply:
I've done the change. Is-it what you expect?

Reply:
I suggest to hide **all** assertions in new custom method. Then we have one meaningful assertion in test, so it becomes easier to read and understand.

Reply:
I've refactored the test. It should be ok.